src/components/import/DropZone.tsx

```diff
@@ -0,0 +1,175 @@
+import ApiClient from '../../services/ApiClient'
+import { FileObj, UploadStatus } from '../../types/FileUpload'
+import './Import.css'
+import UploadFileListItem from './UploadFileListItem'
+
+const DropZone: FC = () => {
+  const [fileObjs, setFilesObj] = useState<FileObj[]>([])
+
+  const onDrop = (
+    acceptedFiles: File[],
+    fileRejections: FileRejection[],
+    event: DropEvent
+  ): void => {
+    setFilesObj([
+      ...fileObjs,
+      { file: acceptedFiles[0], progress: 0, status: UploadStatus.Pending }
+    ])
+  }
+
+  const {
+    acceptedFiles,
+    getRootProps,
+    getInputProps,
+    isFocused,
+    isDragAccept,
+    isDragReject
+  } = useDropzone({
+    accept: { 'text/csv': ['.csv'] },
+    onDrop,
+    maxFiles: 5,
+    multiple: true
+  })
+
+  const uploadFile = async (fileObj: FileObj) => {
+    return await ApiClient.uploadFile(createFileUploadAxiosConfig(fileObj))
+  }
+
+  const createFileUploadAxiosConfig = (
+    fileObj: FileObj
+  ): AxiosRequestConfig<FormData> => {
+    const formData = new FormData()
+    formData.append('file', fileObj.file)
+    return {
+      headers: {
+        'content-type': 'multipart/form-data'
+      },
+      data: formData,
+      onUploadProgress: (progressEvent: AxiosProgressEvent) => {
+        const progress = (progressEvent.loaded / progressEvent.total!) * 100
+        updateFileUploadProgress(fileObj, progress)
+      }
+    }
+  }
+
+  const updateFileUploadProgress = (
+    fileUploadObj: FileObj,
+    progress: number
+  ) => {
+    const fileUploadObjNewState = fileObjs.map(fileObj => {
+      if (fileObj.file.name === fileUploadObj.file.name) {
+        fileObj.progress = progress
+        fileObj.status = getFileUploadStatus(fileObj)
+      }
+      return fileObj
+    })
+    setFilesObj(fileUploadObjNewState)
+  }
+
+  const setUploadStatus = (fileUploadObj: FileObj, status: UploadStatus) => {
+    const fileUploadObjNewState = fileObjs.map(fileObj => {
+      if (fileObj.file.name === fileUploadObj.file.name) {
+        fileObj.status = status
+        if (status === UploadStatus.Failed) {
+          fileObj.progress = 0
+        }
+      }
+      return fileObj
+    })
+    setFilesObj(fileUploadObjNewState)
+  }
+
+  const getFileUploadStatus = (fileObj: FileObj) => {
+    if (fileObj.progress === 0) {
+      return UploadStatus.Pending
+    } else if (fileObj.progress > 0 && fileObj.progress < 100) {
+      return UploadStatus.Loading
+    } else if (fileObj.progress === 100) {
+      return UploadStatus.Complete
+    } else {
+      return UploadStatus.Failed
+    }
+  }
+
+  const uploadFileMutation = useMutation({
+    mutationFn: uploadFile,
+    onSuccess: (data, fileObj) => {
+      setUploadStatus(fileObj, UploadStatus.Complete)
+    },
+    onError: (error: AxiosError, data) => {
+      console.log(`Error uploading files: ${error.message}`)
+      setUploadStatus(data, UploadStatus.Failed)
+    }
```

Comment:
We could use the snackbar in order to show upload success or failure message to the user.